### ISSUE TITLE
cudnnModule.hpp:70:73: error: converting to ‘std::tuple<int, int, int>’ from initializer list...

### ISSUE THREAD

**Issue by lfdmn** [arrayfire/arrayfire#2788]
I get this error when trying to build ArrayFire 3.7

```
In file included from /.../arrayfire_3.7/src/backend/cuda/cudnn.hpp:13:0,
                 from /.../arrayfire_3.7/src/backend/cuda/convolve.cpp:15:
/.../arrayfire_3.7/src/backend/cuda/cudnnModule.hpp: In member function ‘std::tuple<int, int, int> cuda::cudnnModule::getVersion()’:
/.../arrayfire_3.7/src/.../cudnnModule.hpp:70:73: error: converting to ‘std::tuple<int, int, int>’ from initializer list would use explicit constructor ‘constexpr std::tuple< <template-parameter-1-1> >::tuple(_UElements&& ...) [with _UElements = {int&, int&, int&}; <template-parameter-2-2> = void; _Elements = {int, int, int}]’
     std::tuple<int, int, int> getVersion() { return {major, minor, patch}; }
```

Environment: 
- Arrayfire 3.7
- CUDA 9.2
- cuDNN 7.6.5
- boost 1.72
- gcc 5.4.0

**Comment by lfdmn** [comment]
Problem solved by upgrading GCC to 7.4. Please update the [general dependencies for Linux](https://github.com/arrayfire/arrayfire/wiki/Build-Instructions-for-Linux)

For Ubuntu 16.04 based distributions, the latest GCC from package manager is 5.4. So I had to compile and install GCC manually. 

I posted some instructions [https://lfdm.net/2020/03/10/compile-and-install-gcc-on-linux/](https://lfdm.net/2020/03/10/compile-and-install-gcc-on-linux/)

**Comment by 9prady9** [comment]
@lfdmn Sorry about the issue. Thank you for reporting it. 

We moved to C++14 standard for non-API source code. We will look into it and update build instructions accordingly.
